graveler/graveler.go

```diff
@@ -915,55 +920,66 @@ func newStagingToken(repositoryID RepositoryID, branchID BranchID) StagingToken
 	return StagingToken(v)
 }
 
-func (g *Graveler) CommitExistingMetaRange(ctx context.Context, repositoryID RepositoryID, parentCommitID CommitID, metaRangeID MetaRangeID, committer string, message string, metadata Metadata) (CommitID, error) {
-	repo, err := g.RefManager.GetRepository(ctx, repositoryID)
-	if err != nil {
-		return "", fmt.Errorf("get repository %s: %w", repositoryID, err)
-	}
-
-	if parentCommitID != "" {
-		_, err = g.RefManager.GetCommit(ctx, repositoryID, parentCommitID)
+func (g *Graveler) CommitExistingMetaRange(ctx context.Context, repositoryID RepositoryID, branchID BranchID, parentCommitID CommitID, metaRangeID MetaRangeID, committer string, message string, metadata Metadata) (CommitID, error) {
```

Comment:
Think we can use AddCommitNoLock to implement this one.
Consider passing commit record and not all the details.